server/service/src/static_files.rs

```diff
@@ -56,6 +65,25 @@ impl StaticFileService {
         })
     }
 
+    // Temp file in this case refers to system 'TempFile' not our own definition of Temporary file
+    // at the time of method creation TempFile only comes from web multipart
+    pub fn move_temp_file(
+        &self,
+        temp_file: TempFile,
+        category: &StaticFileCategory,
+        file_id: Option<String>,
+    ) -> anyhow::Result<StaticFile> {
+        let file_name = temp_file.file_name.context("Filename not provided")?;
+        let sanitized_filename = sanitize_filename(file_name);
+
+        let static_file = self.reserve_file(&sanitized_filename, &category, file_id)?;
+        let destination = Path::new(&static_file.path);
+        // Is this blocking ? If it is it a problem ?
```

Comment:
Yes I think it is blocking but I don't think it will be a big deal with the filesizes we're dealing with...